### DISCUSSION TITLE
Since when is the nodejs14.x runtime deprecated?

### DISCUSSION THREAD

**Discussion by mariopaglia** [aws/aws-cdk#29108] [Q&A]
Last week I was able to create Lambdas and update using the nodejs14.x runtime normally, this week I can no longer do so, was it a recent update that made this deprecated?

Below is how the error appears:

![image](https://github.com/aws/aws-cdk/assets/18013889/cb12a913-0d03-4cde-9b81-c409a9b64580)

**Comment by iann0036** [comment]
Per https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html#runtime-support-policy, the Node.js 14 runtime was formally deprecated on Dec 4, 2023, with function creation being limited from Jan 9, 2024 and function updates being limited from Feb 8, 2024.

If you have a deprecated runtime within your environment, AWS will send an email notification to the account's primary contact, add a notification in the AWS Health Dashboard, and note it in Trusted Advisor checks.

I recommend you update to a newer Node.js runtime (`node20.x` is latest) and re-test your functions.
